lib/transaction.js

```diff
@@ -17,27 +18,45 @@ var Utils = require('./utils')
  * @param {String} options.isolationLevel=true Sets the isolation level of the transaction.
  * @param {String} options.deferrable Sets the constraints to be deferred or immediately checked.
  */
-var Transaction = module.exports = function(sequelize, options) {
+var Transaction = module.exports = function (sequelize, options) {
   this.sequelize = sequelize;
   this.savepoints = [];
   this.options = Utils._.extend({
     autocommit: true,
     type: sequelize.options.transactionType,
     isolationLevel: sequelize.options.isolationLevel
   }, options || {});
+  if (sequelize.options.dialect !== 'mssql') {
 
-  this.parent = this.options.transaction;
-  this.id = this.parent ? this.parent.id : uuid.v4();
+    this.parent = this.options.transaction;
+    this.id = this.parent ? this.parent.id : uuid.v4();
```

Comment:
Am I missing something or is line 32 redundant? It appears that the conditional on line 34 immediately overwrites what was done on line 32.